importApp/src/test/java/com/commercetools/dataimport/joyrideavailability/AvailabilityPricesImportJobIntegrationTest.java

```diff
@@ -0,0 +1,141 @@
+        final Product product = createProduct(sphereClient);
+        final Long productsCount = 1L;
+        assertThat(findLastProductWithJoyrideChannel(sphereClient, 0L, productsCount - 1)).isEmpty();
+        withPriceWithJoyrideChannel(sphereClient, priceDraft -> {
+            final AddPrice addPrice = AddPrice.of(product.getMasterData().getCurrent().getMasterVariant().getId(), priceDraft);
+            Product updatedProduct = sphereClient.executeBlocking(ProductUpdateCommand.of(product, asList(addPrice, Publish.of())));
+            final Optional<ProductProjection> lastProductWithJoyrideChannel = findLastProductWithJoyrideChannel(sphereClient, 0L, productsCount - 1);
+            assertThat(lastProductWithJoyrideChannel).isPresent();
+            assertThat(lastProductWithJoyrideChannel.get().getMasterVariant().getSku()).isEqualTo(product.getMasterData().getCurrent().getMasterVariant().getSku());
+            final Product productToDelete = sphereClient.executeBlocking(ProductUpdateCommand.of(updatedProduct, Unpublish.of()));
+            sphereClient.executeBlocking(ProductDeleteCommand.of(productToDelete));
+        });
+    }
+
+    @Test
+    public void jobAvailabilityPricesImport() throws Exception {
+        final int amountOfProducts = 10;
+        withJoyrideChannels(sphereClient, joyrideChannels -> {
+            withListOfProductProjections(sphereClient, amountOfProducts, productsWithoutJoyride -> {
+                assertThat(productsWithoutJoyride.size()).isEqualTo(amountOfProducts);
+                executeAvailabilityPricesImportJob();
+                final List<ProductProjection> updatedProducts = sphereClient.executeBlocking(ProductProjectionQuery.ofCurrent().withExpansionPaths(m -> m.allVariants().prices().channel())).getResults();
+                updatedProducts.forEach(m -> {
+                    assertThat(productContainJoyridePrice(m)).isTrue();
+                    final ProductProjection productProjectionWithoutJoyride = productsWithoutJoyride.stream().filter(p -> p.getId().equals(m.getId())).limit(1L).collect(toList()).get(0);
+                    final Price originalMasterVariantPrice = productProjectionWithoutJoyride.getMasterVariant().getPrices().get(0);
+                    final ProductVariant masterVariant = m.getMasterVariant();
+                    masterVariant.getPrices().forEach(variantPrice -> {
+                        if (variantPrice.getChannel() != null) {
+                            final PriceDraftDsl randomPriceDraft = randomPriceDraft(masterVariant, variantPrice.getChannel().getObj(), originalMasterVariantPrice);
+                            assertThat(variantPrice.getValue()).isEqualTo(randomPriceDraft.getValue());
+                        }
+                    });
+                });
+                return updatedProducts;
+            });
+        });
+    }
+
+    @Test
+    public void readerRestartAfterPartialExecution() throws Exception {
```

Comment:
This test is quite complicated, I would expect that in the first case you just assert that the reader starts from the beginning and then the next reader, after data import stopped in the middle, then returns the product where to go on.